src/summary_statistics/means.rs

```diff
@@ -36,15 +37,163 @@ where
+    where
+        A: Float + FromPrimitive,
+    {
+        let central_moments = self.central_moments(4)?;
+        Ok(central_moments[4] / central_moments[2].powi(2))
+    }
+
+    fn skewness(&self) -> Result<A, EmptyInput>
+    where
+        A: Float + FromPrimitive,
+    {
+        let central_moments = self.central_moments(3)?;
+        Ok(central_moments[3] / central_moments[2].sqrt().powi(3))
+    }
+
+    fn central_moment(&self, order: u16) -> Result<A, EmptyInput>
+    where
+        A: Float + FromPrimitive,
+    {
+        if self.is_empty() {
+            return Err(EmptyInput);
+        }
+        match order {
+            0 => Ok(A::one()),
+            1 => Ok(A::zero()),
+            n => {
+                let mean = self.mean().unwrap();
+                let shifted_array = self.mapv(|x| x - mean);
+                let shifted_moments = moments(shifted_array, n);
+                let correction_term = -shifted_moments[1];
+
+                let coefficients = central_moment_coefficients(&shifted_moments);
+                Ok(horner_method(coefficients, correction_term))
+            }
+        }
+    }
+
+    fn central_moments(&self, order: u16) -> Result<Vec<A>, EmptyInput>
+    where
+        A: Float + FromPrimitive,
+    {
+        if self.is_empty() {
+            return Err(EmptyInput);
+        }
+        match order {
+            0 => Ok(vec![A::one()]),
+            1 => Ok(vec![A::one(), A::zero()]),
+            n => {
+                // We only perform these operations once, and then reuse their
+                // result to compute all the required moments
+                let mean = self.mean().unwrap();
+                let shifted_array = self.mapv(|x| x - mean);
+                let shifted_moments = moments(shifted_array, n);
+                let correction_term = -shifted_moments[1];
+
+                let mut central_moments = vec![A::one(), A::zero()];
+                for k in 2..=n {
+                    let coefficients =
+                        central_moment_coefficients(&shifted_moments[..=(k as usize)]);
+                    let central_moment = horner_method(coefficients, correction_term);
+                    central_moments.push(central_moment)
+                }
+                Ok(central_moments)
+            }
+        }
+    }
+}
+
+/// Returns a vector containing all moments of the array elements up to
+/// *order*, where the *p*-th moment is defined as:
+///
+/// ```text
+/// 1  n
+/// ―  ∑ xᵢᵖ
+/// n i=1
+/// ```
+///
+/// The returned moments are ordered by power magnitude: 0th moment, 1st moment, etc.
+///
+/// **Panics** if `A::from_usize()` fails to convert the number of elements in the array.
+fn moments<A, S, D>(a: ArrayBase<S, D>, order: u16) -> Vec<A>
+where
+    A: Float + FromPrimitive,
+    S: Data<Elem = A>,
+    D: Dimension,
+{
+    let n_elements =
+        A::from_usize(a.len()).expect("Converting number of elements to `A` must not fail");
+    let order = order
```

Comment:
Now that `order` is `u16`, we can remove this.

Reply:
True, it can't fail anymore. I'll merge once the CI has finished :+1: